jdi-light/src/main/java/com/epam/jdi/light/elements/base/JDIBase.java

```diff
@@ -200,35 +203,47 @@ private UIElement getUIElementByLocator(WebElement element) {
     public WebElement get(Object... args) {
         manageTimeout();
         if (webElement.hasValue()) {
-            WebElement element = purify(webElement.get());
-            try {
-                element.getTagName();
-                beforeSearch(element);
-                return !locator.isEmpty()
-                    ? purify(getUIElementByLocator(element))
-                    : element;
-            } catch (Exception ignore) {
-                if (getElementFunc == null)
-                    webElement.clear();
-                else {
-                    return beforeSearch(webElement.set(purify(getElementFunc.execute())));
-                }
-            }
+            WebElement webElement = getCachedElement();
+            if (webElement != null)
+                return webElement;
         }
         if (locator.isEmpty())
             return beforeSearch(getSmart());
-        if (locator.argsCount() != args.length) {
-            if (locator.argsCount() == 0 && args.length == 1) {
-                if (args[0].getClass() == String.class)
-                    return new WebList(this).get(args[0].toString());
-                if (isClass(args[0].getClass(), Enum.class))
-                    return new WebList(this).get(getEnumValue((Enum)args[0]));
-            }
-            throw exception("Can't get element with template locator '%s'. Expected %s arguments but found %s", getLocator(), locator.argsCount(), args.length);
-        }
+        if (locator.argsCount() != args.length)
+            return getWebListFromArgs(args);
         List<WebElement> els = getAllElements(args);
         return getElement(els);
     }
+    private UIElement getWebListFromArgs(Object... args) {
+        if (locator.argsCount() == 0 && args.length == 1) {
+            if (args[0].getClass() == String.class)
+                return new WebList(this).get(args[0].toString());
+            if (isClass(args[0].getClass(), Enum.class))
+                return new WebList(this).get(getEnumValue((Enum<?>)args[0]));
+        }
+        throw exception("Can't get element with template locator '%s'. Expected %s arguments but found %s", getLocator(), locator.argsCount(), args.length);
+    }
+    private WebElement getCachedElement() {
+        WebElement element = purify(webElement.get());
+        try {
+            element.getTagName();
+            beforeSearch(element);
+            return !locator.isEmpty()
+                    ? purify(getUIElementByLocator(element))
+                    : element;
+        } catch (Exception ignore) {
+            if (getElementFunc == null) {
+                webElement.clear();
+                return null;
+            } else {
+                return beforeSearch(webElement.set(purify(getElementFunc.execute())));
+            }
+        }
+    }
+    public Boolean strictSearch;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5217753)